### PR TITLE
[adaptor] APIPolicy ref to InterceptorService kind changes

### PR DESCRIPTION
**Pull request by Amila-Rukshan**
## Purpose
$subject and added multiple interceptors entry support in APIPolicy CRD, only the first entry is read in both request and response interceptors paths.

Sample resources with updated configuration
```
apiVersion: dp.wso2.com/v1alpha1
kind: APIPolicy
metadata:
  name: sample-api-policy
  namespace: interceptor
spec:
  override:
    requestInterceptors:
    - name: interceptor-service-3
      namespace: interceptor
    responseInterceptors:
    - name: interceptor-service-4
      namespace: interceptor
  targetRef:
    group: gateway.networking.k8s.io
    kind: API
    name: interceptor-api
    namespace: interceptor
---
apiVersion: dp.wso2.com/v1alpha1
kind: InterceptorService
metadata:
  name: interceptor-service-3
  namespace: interceptor
spec:
  backendRef:
    name: interceptor-backend-2
  includes:
  - request_body
  - request_headers
  - request_trailers
  - invocation_context
---
apiVersion: dp.wso2.com/v1alpha1
kind: InterceptorService
metadata:
  name: interceptor-service-4
  namespace: interceptor
spec:
  backendRef:
    name: interceptor-backend-2
  includes:
  - response_headers
  - response_body
```

## Tested
Gateway, API, Resource levels with the new changes

### CODE REVIEW

File: adapter/internal/operator/apis/dp/v1alpha1/interceptorservice_types.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/*
+ *  Copyright (c) 2022, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (AmaliMatharaarachchi) [inline adapter/internal/operator/apis/dp/v1alpha1/interceptorservice_types.go:2]:
```suggestion
 *  Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```

File: adapter/pkg/logging/logging_constant.go [hunk 2]
```diff
@@ -339,6 +348,16 @@ var Mapper = map[int]ErrorDetails{
 		Message:   "Error while getting custom rate limit policies: %s",
 		Severity:  MAJOR,
 	},
+	error2651: {
+		ErrorCode: error2651,
+		Message:   "Error while getting interceptor service %s, %s",
+		Severity:  BLOCKER,
```
Comment (AmaliMatharaarachchi) [inline adapter/pkg/logging/logging_constant.go:354]: is this a blocking error?

Comment (Amila-Rukshan) [reply adapter/pkg/logging/logging_constant.go:354]: this error is when a api server connectivity or its internal error scenario, so kept as a blocker

**Review by AmaliMatharaarachchi — APPROVED** [review]

File: adapter/pkg/logging/logging_constant.go (hunk 2 above)
Comment (Amila-Rukshan) [reply adapter/pkg/logging/logging_constant.go]: changed to `CRITICAL` severity level

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Pull request merged**